packages/swap/src/routes/v2/swap.ts

```diff
@@ -45,16 +45,23 @@ router.get(
       swapEgress,
       refundEgress,
       ignoredEgresses,
+      vaultSwap,
     } = await getLatestSwapForId(id);
 
     const { state, swapEgressTrackerTxRef, refundEgressTrackerTxRef, pendingDeposit } =
-      await getSwapState(failedSwap, ignoredEgresses, swapRequest, swapDepositChannel);
-
-    const internalSrcAsset = readField(swapRequest, swapDepositChannel, failedSwap, 'srcAsset');
-    const internalDestAsset = readField(swapRequest, swapDepositChannel, failedSwap, 'destAsset');
+      await getSwapState(failedSwap, ignoredEgresses, swapRequest, swapDepositChannel, vaultSwap);
+
+    const internalSrcAsset =
+      readField(swapRequest, swapDepositChannel, failedSwap, 'srcAsset') ??
+      (vaultSwap && getInternalAsset(vaultSwap.inputAsset as UncheckedAssetAndChain));
```

Comment:
we could transform the `vaultSwap` in `getLatestSwapForId` already so we dont need to do this multiple times. but not a strong opinion